app/lib/search/mem_index.dart

```diff
@@ -19,15 +20,15 @@ final _logger = Logger('search.mem_index');
 final _textSearchTimeout = Duration(milliseconds: 500);
 
 class InMemoryPackageIndex {
-  final Map<String, PackageDocument> _packages = <String, PackageDocument>{};
-  final _packageNameIndex = PackageNameIndex();
+  final Map<String, PackageDocument> _packages = {};
+  final PackageNameIndex _packageNameIndex = PackageNameIndex();
   final TokenIndex _descrIndex = TokenIndex();
   final TokenIndex _readmeIndex = TokenIndex();
   final TokenIndex _apiSymbolIndex = TokenIndex();
 
   /// Adjusted score takes the overall score and transforms
   /// it linearly into the [0.4-1.0] range.
-  final _adjustedOverallScores = <String, double>{};
+  final Map<String, double> _adjustedOverallScores = {};
```

Comment:
Same here: type inference seems to be enough, why the extra specification?

Reply:
The inference is enough, but this change allows it to be consistent with the other nearby field entries.

Since these are fields and can be used outside of the direct vicinity, I generally think it makes sense to specify (and as a result, think about) the type.

Reply:
Oh, now I understand the intent here. The other nearby field entries are older code that preceded the inference capabilities in the language, so if we wanted to make them consistent, I'd rather move them towards using the more modern features. wdyt?